tuf/api/metadata.py

```diff
@@ -417,6 +417,7 @@ class Key:
     """A container class representing the public portion of a Key.
 
     Attributes:
+        id: An identifier string
```

Comment:
"An identifier string" is true from the Key class perspective but I think some hints about it being the digest of  an ambiguous canonical key is needed.

Reply:
yeah, makes sense -- I'm not fond of documenting the (probably untrue) idea that we could calculate the id from the key content but I should document that the id must be unique within the keys in this Metadata

Reply:
```
        id: An identifier string that must uniquely identify a key within
            the metadata it is used in. This implementation does not verify
            that the id is the hash of a specific representation of the key.
```
this work for you?

Reply:
works for me :)